packages/webpack/lib/setEnv.js

```diff
@@ -0,0 +1,40 @@
+module.exports = setEnv
+
+/**
+ * Replaces constants in your source code with values from `process.env`
+ * using the `webpack.EnvironmentPlugin`.
+ *
+ * Using `setEnv` multiple times results in a single
+ * EnvironmentPlugin instance configured to do all the replacements.
+ *
+ * @param {string[]|object} constants
+ * @return {Function}
+ */
+function setEnv (constants) {
+  const setter = context => prevConfig => {
+    context.setEnv = Object.assign({}, context.setEnv, toObject(constants))
+    return prevConfig
+  }
+
+  return Object.assign(setter, { post: addEnvironmentPlugin })
+}
+
+function addEnvironmentPlugin (context, util) {
+  return util.addPlugin(
+    new context.webpack.EnvironmentPlugin(context.setEnv)
+  )
+}
+
+function toObject (constants) {
+  if (Array.isArray(constants)) {
+    return constants.reduce((result, constant) => {
+      // Setting the default (fallback) value to `undefined`
```

Comment:
👍

Reply:
The thing is all these `undefined`s and `null`s are resolved in DefinePlugin, and yes, it lacks documentation like many other things, but I don't believe it's an excuse to introduce a breaking change in a minor release for them.